src/cdk/v2/destinations/fullstory/procWorkflow.yaml

```diff
@@ -0,0 +1,104 @@
+bindings:
+  - name: EventType
+    path: ../../../../constants
+  - path: ../../bindings/jsontemplate
+    exportAll: true
+  - name: removeUndefinedAndNullValues
+    path: ../../../../v0/util
+  
+steps:
+  - name: validateInput
+    template: |
+      $.assert(.message.type, "message Type is not present. Aborting message.");
+      $.assert(.message.type in {{$.EventType.([.TRACK, .IDENTIFY])}}, 
+        "message type " + .message.type + " is not supported");
+  - name: prepareContext
+    template: |
+      $.context.messageType = .message.type.toLowerCase();
+      $.context.payload = {};
+      $.context.finalHeaders = {
+          "authorization": "Basic " + .destination.Config.apiKey,
+          "content-type": "application/json"
+        };
+  - name: identifyPayload
+    condition: $.context.messageType == "identify"
+    template: |
+      $.context.endpoint = "https://api.fullstory.com/v2/users";
+      $.context.payload.properties = .message.traits ?? .message.context.traits;
+      $.context.payload.uid = .message.userId;
+      $.context.payload.email = .message.context.traits.email;
+      $.context.payload.display_name = .message.context.traits.name;
+  
+  - name: trackPayload
+    condition: $.context.messageType == "track"
+    template: |
+      $.context.endpoint = "https://api.fullstory.com/v2/events";
```

Comment:
[Q] Does fullstory support different endpoints for EU or US instances ? If we hadn't verified it, can we quickly verify it ?

Reply:
Yes, but it is handled from fullstory's end.
As per their [data residency docs](https://developer.fullstory.com/server/getting-started/#data-residency),

> All API requests use the api.fullstory.com domain. They are automatically routed to the correct data center based on what is configured for your FullStory account. You can determine which data center your account is associated with in one of two ways:
> - The FullStory URL you use to login
> - API key format: <data center>.<token>